packages/playwright/src/AxeCoreWrapper/DefaultAxeCoreWrapper.cs

```diff
@@ -1,108 +1,297 @@
+            }
+
+            return results;
+        }
+
+        /// <summary>
+        /// Runs axe via legacyScan.js at a specific context, which will be passed as-is to Selenium for legacyScan.js to interpret, and
+        /// parses/handles the legacyScan.js output per the current builder options.
+        /// </summary>
+        /// <param name="rawContextArg">The value to pass as-is to legacyScan.js to use as the axe.run "context" argument</param>
+        private async Task<AxeResult> RunInner(IFrame frame, object rawContextArg, AxeRunOptions? runOptions)
+        {
+            await ConfigureAxe(frame);
+
+
+            if (runOptions == null)
+            {
+                runOptions = new AxeRunOptions
+                {
+                    Iframes = true
+                };
+            }
+
+            var runPartialExists = await RunPartialExists(frame);
+
+            if (!runPartialExists)
+            {
+                await m_axeContentEmbedder.EmbedAxeCoreIntoPage(frame.Page, runOptions.Iframes, true);
+                return await EvaluateAxeRun(frame.Page, null, runOptions);
+            }
+            else
+            {
+                return await AnalyzeAxeRunPartial(frame, rawContextArg, runOptions);
+            }
+
+
+        }
+
+        private async Task<AxeResult> AnalyzeAxeRunPartial(IFrame frame, object rawContextArg, AxeRunOptions runOptions)
+        {
+            string rawOptionsArg = JsonConvert.SerializeObject(runOptions);
+            var partialResults = await RunPartialRecursive(frame, rawOptionsArg, rawContextArg, true, !runOptions.Iframes.HasValue || runOptions.Iframes.Value);
+            return await IsolatedFinishRun(frame, partialResults.ToArray(), rawOptionsArg);
+        }
+
+        private async Task<List<object?>> RunPartialRecursive(
+                                                        IFrame frame,
+                                                        object options,
+                                                        object context,
+                                                        bool isTopLevel,
+                                                        bool iframes
+                                                        )
+        {
+            if (!isTopLevel)
+            {
+                await ConfigureAxe(frame);
+            }
+
+            var partialResults = new List<object?>();
+
+            try
+            {
+                string partialRes = await frame.EvaluateAsync<string>(EmbeddedResourceProvider.ReadEmbeddedFile("runPartial.js"), new[] { context, options });
+                // Important to deserialize because we want to reserialize as an
+                // array of object, not an array of strings.
+                partialResults.Add(JsonConvert.DeserializeObject<object>(partialRes));
+            }
+            catch (Exception ex)
+            {
+                if (isTopLevel)
+                {
+                    throw ex;

```

Comment:
In C#, you generally want to prefer `throw;` instead of `throw ex;` for cases where you want a catch block to conditionally propagate the originally thrown exception. The former will maintain the original stack track, which is probably what you want here, where the latter will reset the stack trace to this line and lose the inner context about where the exception was thrown.

Reply:
Good to know.

Reply:
The .NET SDK we're using actually has a static analyzer that can warn about this automatically, but it looks like it's not running because we (intentionally) target a downlevel .NET standard version. I'll create a separate PR to enable the static analyzers to catch this automatically in the future.

Reply:
#66